packages/osd-config/src/env.ts

```diff
@@ -138,6 +147,7 @@ export class Env {
     this.cliArgs = Object.freeze(options.cliArgs);
     this.configs = Object.freeze(options.configs);
     this.isDevClusterMaster = options.isDevClusterMaster;
+    this.isDevClusterManager = options.isDevClusterManager;
```

Comment:
I don't want to add too many cooks to this, because this PR is in fine shape, but I think it's worthwhile for us as a team to have a good playbook for deprecating public interfaces, because this isn't the only time we're going to have to do this (and we will still have more PRs for non-inclusive language). But generally the approach I'd favor is 

1. Add the new external interfaces (`isDevClusterManager`, in this case)
2. Mark the old interfaces deprecated (`isDevClusterMaster`)
3. Make all _internal_ properties read from both, with a priority of the new interface (`this.isDevClusterManager = options.isDevClusterManager || options.isDevClusterMaster`). That way the fallback logic is not spread throughout the codebase.
4. Duplicate any existing tests of the old interface for the new one, so that, until we deprecate, both are covered independently. (Depending on the type of deprecation, we may also want to test that the priority order from 3. works correctly if both interfaces are used and they conflict.)

We're already doing steps 1 and 2, so the difference is in 3 and 4. There's an argument that this approach may make the eventual removal/cleanup more difficult, but I think it's a bit of a toss up. Curious for the rest of the team's perspective on the above.

For this purpose of this PR, it would mean  changing this line, removing subsequent fallbacks (`!(this.coreContext.env.isDevClusterMaster || this.coreContext.env.isDevClusterManager);`) and adding additional tests.

Reply:
My initial thought with this was if I  have a new interface field introduced without having to touch the old one, felt more safer option to not break things for the downstream dependencies which might be consuming the old field as it is! But thinking through it and realizing the 'internal' tag on the internal properties, it make sense to go with 3rd point to be more clean and precise. Good call out on point 4 to include tests for the new interface property added. I can update the PR. 

> For this purpose of this PR, it would mean changing this line, removing subsequent fallbacks (!(this.coreContext.env.isDevClusterMaster || this.coreContext.env.isDevClusterManager);) and adding additional tests.
>
@joshuarrrr I think you mean changing the order of the fields checked here, right?